GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -542,38 +545,41 @@ private void InternalInitialize(bool hard)
                     updateForm.SearchForUpdatesAndShow(Owner, false);
                 }
 
-                bool bareRepository = Module.IsBareRepository();
-                bool validWorkingDir = Module.IsValidGitWorkingDir();
                 bool hasWorkingDir = !string.IsNullOrEmpty(Module.WorkingDir);
-                branchSelect.Text = validWorkingDir ? Module.GetSelectedBranch() : "";
-                if (hasWorkingDir)
+                if (hasWorkingDir && !_startWithDashboard)
                 {
                     HideDashboard();
                 }
                 else
                 {
+                    // Consume the startup arguments, no specific directory was requested
+                    _startWithDashboard = false;
                     ShowDashboard();
                 }
 
+                bool bareRepository = Module.IsBareRepository();
+                bool isDashboard = _dashboard != null && _dashboard.Visible;
+                bool validBrowseDir = Module.IsValidGitWorkingDir() && !isDashboard;
```

Comment:
I would invert it, checking for `!isDashboard` is faster than executing the function

Reply:
!isDashboard is almost always true (if we start with dashboard it is false once).
IsValidGitWorkingDir() is almost always true too.
So no performance change that we care about.
Can change.